haystack/nodes/reader/table.py

```diff
@@ -241,76 +243,68 @@ def __init__(
             model_name_or_path, revision=model_version, use_auth_token=use_auth_token
         )
         if tokenizer is None:
-            self.tokenizer = TapasTokenizer.from_pretrained(model_name_or_path, use_auth_token=use_auth_token)
+            self.tokenizer = TapasTokenizer.from_pretrained(
+                model_name_or_path, use_auth_token=use_auth_token, model_max_length=max_seq_len
+            )
         else:
-            self.tokenizer = TapasTokenizer.from_pretrained(tokenizer, use_auth_token=use_auth_token)
+            self.tokenizer = TapasTokenizer.from_pretrained(
+                tokenizer, use_auth_token=use_auth_token, model_max_length=max_seq_len
+            )
         self.max_seq_len = max_seq_len
         self.device = device
+        self.pipeline = _TableQuestionAnsweringPipeline(
+            task="table-question-answering",
+            model=self.model,
+            tokenizer=self.tokenizer,
+            framework="pt",
+            batch_size=1,  # batch_size of 1 only works currently b/c of issue with HuggingFace pipeline logic and the return type of TableQuestionAnsweringPipeline._forward
+            device=self.device,
+        )
 
-    def _predict_tapas(self, inputs: BatchEncoding, document: Document) -> Answer:
-        orig_table: pd.DataFrame = document.content
-        string_table = orig_table.astype(str)
+    def predict(self, query: str, documents: List[Document], top_k: int) -> Dict:
+        table_documents = _check_documents(documents)
+        if len(table_documents) == 0:
+            return {"query": query, "answers": []}
 
-        # Forward query and table through model and convert logits to predictions
-        self.model.eval()
-        with torch.inference_mode():
-            outputs = self.model(**inputs)
+        # Create list of all data points
+        pipeline_inputs = []
```

Comment:
Should we move these 4 lines to a separate function like below?

```
def _docs_to_tableQA_pipeline_input(table_documents):
    pipeline_inputs = []
    for document in table_documents:
        table: pd.DataFrame = document.content
        pipeline_inputs.append({"query": query, "table": table.astype(str)})
    return pipeline_inputs
```

Reply:
I don't think this is currently necessary because this function would not be used anywhere else and I think the current version of the `predict` function is not too long or complicated. If we ever need to reuse this code elsewhere then I think making it its own function is a good idea.